src/schema.rs

```diff
@@ -302,6 +303,56 @@ impl Team {
         Ok(lists)
     }
 
+    pub(crate) fn zulip_groups(&self, data: &Data) -> Result<Vec<ZulipGroup>, Error> {
+        let mut groups = Vec::new();
+        let zulip_groups = match &self.zulip_groups {
+            Some(zgs) => zgs,
+            None => return Ok(Vec::new()),
+        };
+
+        for raw_group in zulip_groups {
+            let mut list = ZulipGroup {
+                name: raw_group.name.clone(),
+                members: Vec::new(),
+            };
+
+            let mut members = if raw_group.include_team_members {
+                self.members(data)?
+            } else {
+                HashSet::new()
+            };
+            for person in &raw_group.extra_people {
+                members.insert(person.as_str());
+            }
+            for team in &raw_group.extra_teams {
+                let team = data
+                    .team(team)
+                    .ok_or_else(|| err_msg(format!("team {} is missing", team)))?;
+                for member in team.members(data)? {
+                    members.insert(member);
+                }
+            }
+
+            for member in members.iter() {
+                let member = data
+                    .person(member)
+                    .ok_or_else(|| err_msg(format!("member {} is missing", member)))?;
+                if let Some(zulip_id) = member.zulip_id {
+                    list.members.push(ZulipGroupMember::Id(zulip_id));
+                }
+                if let Email::Present(email) = member.email() {
+                    list.members
+                        .push(ZulipGroupMember::Email(email.to_string()));
+                }
```

Comment:
We should probably add either the ID or the email (preferring the ID if present), not both. It's more intuitive to me that if an explicit ID is present it's going to be used rather than the email address. There could also be a *slight* chance that the ID and the email map to different accounts, which would be weird.